apps/group/contracts/Group.sol

```diff
@@ -45,26 +65,78 @@ contract Group is App, Initializable, IForwarder, EVMCallScriptRunner {
     */
     function removeMember(address _entity) auth(REMOVE_MEMBER_ROLE) external {
         require(isGroupMember(_entity));
-        members[_entity] = false;
+        isMembers[_entity] = false;
+        for (uint i = 0; i < members.length; i++) {
+            if (members[i] == _entity) {
+                members[i] = members[members.length - 1];
+                delete members[members.length - 1];
+                members.length -= 1;
+                break;
+            }
+        }
+
+        if (required > members.length)
+            changeRequirement(members.length);
+
         RemoveMember(_entity);
     }
 
+    function numMembers() public constant returns (uint) {
+        return members.length;
+    }
+
+    function getRequired() public constant returns (uint) {
+        return required;
+    }
+
+    function confirm() public {
+        require(isGroupMember(msg.sender));
+        confirmations[msg.sender] = true;
```

Comment:
Confirmations should happen on an action per action basis, right now once enough confirmations have been given to the contract, any member can perform any actions at any moment (until confirmations are revoked).

For an example of how this can be achieved you can see [Gnosis' Multisig](https://github.com/gnosis/MultiSigWallet/blob/master/contracts/MultiSigWallet.sol#L30).